packages/quantic/force-app/main/default/lwc/quanticResultPrintableUri/__tests__/quanticResultPrintableUri.test.js

```diff
@@ -0,0 +1,149 @@
+  return element;
+}
+
+// Helper function to wait until the microtask queue is empty.
+function flushPromises() {
+  return new Promise((resolve) => setTimeout(resolve, 0));
+}
+
+describe('c-quantic-result-printable-uri', () => {
+  function cleanup() {
+    // The jsdom instance is shared across test cases in a single file so reset the DOM
+    while (document.body.firstChild) {
+      document.body.removeChild(document.body.firstChild);
+    }
+    jest.clearAllMocks();
+  }
+
+  beforeEach(() => {
+    // @ts-ignore
+    global.Bueno = {
+      isString: jest
+        .fn()
+        .mockImplementation(
+          (value) => Object.prototype.toString.call(value) === '[object String]'
+        ),
+    };
+  });
+
+  afterEach(() => {
+    cleanup();
+  });
+
+  describe('when displaying the url', () => {
+    describe('when the result is of type Salesforce', () => {
+      it('should call the navigation mixin to get the Salesforce record URL', async () => {
+        createTestComponent({
+          ...mockSalesforceResult,
+        });
+        await flushPromises();
+
+        const {pageReference} = getGenerateUrlCalledWith();
+
+        expect(pageReference.attributes.recordId).toBe(
+          mockSalesforceResult.result.raw.sfid
+        );
+      });
+
+      it('should open the result link in a Salesforce subtab', async () => {
+        const element = createTestComponent({
+          ...mockSalesforceResult,
+        });
+        await flushPromises();
+
+        const linkSalesforce = element.shadowRoot.querySelector(selectors.link);
+        linkSalesforce.click();
+
+        const {pageReference} = getNavigateCalledWith();
+
+        expect(pageReference.attributes.recordId).toBe(
+          mockSalesforceResult.result.raw.sfid
+        );
+      });
+
+      describe('when the result is a knowledge article', () => {
+        it('should open the result link in a Salesforce console subtab', async () => {
+          const element = createTestComponent({
+            ...mockKnowledgeArticleResult,
+          });
+          await flushPromises();
+
+          const LinkKnowledgeArticle = element.shadowRoot.querySelector(
+            selectors.link
+          );
+          LinkKnowledgeArticle.click();
+
+          const {pageReference} = getNavigateCalledWith();
+
+          expect(pageReference.attributes.recordId).toBe(
+            mockKnowledgeArticleResult.result.raw.sfkavid
+          );
+        });
+      });
+    });
+
+    describe('when the result is not of type Salesforce', () => {
+      it('should open the result link in a browser tab', async () => {
+        const element = createTestComponent({
+          ...mockDefaultResult,
+        });
+        await flushPromises();
+
+        const link = element.shadowRoot.querySelector(selectors.link);
+
+        expect(link.getAttribute('href')).toEqual(
+          mockDefaultResult.result.printableUri
+        );
+        expect(link.getAttribute('target')).toEqual('_self');
```

Comment:
```suggestion
        const linkDefault = element.shadowRoot.querySelector(selectors.link);

        expect(linkDefault.getAttribute('href')).toEqual(
          mockDefaultResult.result.printableUri
        );
        expect(linkDefault.getAttribute('target')).toEqual('_self');
```

Reply:
I don't think this is really necessary, after all we are only selecting a link. `linkDefault` is not really adding a new information and is not improving the readability of the tests.

Reply:
I thought it would make more sense since we are using the mockDefaultResult for the test, but its your choice.